Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -0,0 +1,35 @@
+class JavaRequirement
```

Comment:
It doesn't seem to make much difference actually. `JavaRequirement.new(["1.8+"]).is_a? Requirement` is true on macOS either way, and the Mac-specific overrides work either way. Will submit anyhow, because at least it's more consistent.